cdap-docs/developers-manual/source/data-exploration/streams.rst

```diff
@@ -0,0 +1,214 @@
+
+End-to-end Example
+------------------
+
+In the following example, we will create a Stream, send data to it, attach a format
+and schema to the stream, then query the Stream.
+
+Suppose we want to create a Stream for stock trades. We first create the stream
+and send some data to it as comma separated text::
+
+  > create stream trades
+  > send stream trades "AAPL,50,112.98"
+  > send stream trades "AAPL,100,112.87"
+  > send stream trades "AAPL,8,113.02"
+  > send stream trades "NFLX,10,437.45"
+
+If we run a query over the Stream, we can see each event as text::
+
+  > execute "select * from cdap_stream_trades"
+  +==================================================================================================================+
+  | cdap_stream_trades.ts: BIGINT | cdap_stream_trades.headers: map<string,string> | cdap_stream_trades.body: STRING |
+  +==================================================================================================================+
+  | 1422493022983                 | {}                                             | AAPL,50,112.98                  |
+  | 1422493027358                 | {}                                             | AAPL,100,112.87                 |
+  | 1422493031802                 | {}                                             | AAPL,8,113.02                   |
+  | 1422493036080                 | {}                                             | NFLX,10,437.45                  |
+  +==================================================================================================================+
+
+Since we know the body of every event is comma separated text and that each event
+contains three fields, we can set a format and schema on the stream to allow us to run more
+complicated queries::
+
+  > set stream format trades csv "ticker string, num_traded int, price double"
+  > execute "select ticker, count(*) as transactions, sum(num_traded) as volume from cdap_stream_trades group by ticker order by volume desc" 
+  +========================================================+
+  | ticker: STRING | transactions: BIGINT | volume: BIGINT |
+  +========================================================+
+  | AAPL           | 3                    | 158            |
+  | NFLX           | 1                    | 10             |
+  +========================================================+
+
+Formulating Queries
+-------------------
+When creating your queries, keep these limitations in mind:
+
+- The query syntax of CDAP is a subset of the variant of SQL that was first defined by Apache Hive.
+- Writing into a Stream using SQL is not allowed.
+- The SQL command ``DELETE`` is not allowed.
+- When addressing your streams in queries, you need to prefix the stream name with the CDAP
+  namespace ``cdap_stream_``. For example, if your Stream is named ``Purchases``, then the corresponding table
+  name is ``cdap_user_purchases``. Note that the table name is lower-case.
```

Comment:
Note that the table name is all lower-case letters, regardless of how it was defined.